recommender/recommender.py

```diff
@@ -999,11 +1023,35 @@ def parse_xml(cls, node, runtime, keys, _id_generator):  # pylint: disable=unuse
 
         """
         block = runtime.construct_xblock_from_class(cls, keys)
-        lines = []
-        for line in node.text.split('\n'):
-            line = line.strip()
-            if len(line) > 2:
-                lines.append(json.loads(line))
 
-        block.default_recommendations = data_structure_upgrade(lines)
+        node_xml = etree.tostring(node, encoding='utf-8')
+        try:
+            dom = md.parseString(node_xml.encode('utf-8'))
+        except Exception:
+            raise UpdateFromXmlError("An error occurred while parsing the XML content.")
+        root = dom.documentElement
+
+        if root.tagName != 'recommender':
+            raise UpdateFromXmlError("XML content must contain an 'recommender' root element.")
+
+        block.intro_enabled = (root.getAttribute('intro_enabled').lower().strip() != "false")
+        block.client_side_settings['DISABLE_DEV_UX'] = (root.getAttribute('DISABLE_DEV_UX').lower().strip() != "false")
+
+        for tag in ['ENTRIES_PER_PAGE', 'PAGE_SPAN']:
+            if root.getAttribute(tag) is not None and len(root.getAttribute(tag)) > 0:
```

Comment:
You can replace this whole like with
```python
if root.getAttribute(tag):
```
Checking for truthy/falsy values is useful and concise!